beancount/utils/misc_utils.py

```diff
@@ -281,15 +282,15 @@ def compute_unique_clean_ids(strings):
     return idmap
 
 
-def escape_string(string):
-    """Escape quotes and backslashes in payee and narration.
+def escape_string(string: str) -> str:
+    """Escape backslashes, double quotes, and control characters.
 
     Args:
       string: Any string.
     Returns.
       The input string, with offending characters replaced.
     """
-    return string.replace("\\", r"\\").replace('"', r"\"")
+    return json.dumps(string, ensure_ascii=False)[1:-1]
```

Comment:
This introduces a backward incompatible change to the serialization code. At the very minimum it needs to be in a separate PR all together, where the merits of the change can be discussed. This change is not required to fix the tests. See #876.